pkg/controller/issueapi/handle_issue.go

```diff
@@ -104,16 +104,16 @@ func (c *Controller) decodeAndIssue(ctx context.Context, w http.ResponseWriter,
 
 	res := c.IssueOne(ctx, &request)
 	result.HTTPCode = res.HTTPCode
-	resp := res.IssueCodeResponse()
-	if resp.Error != "" {
-		if result.HTTPCode == http.StatusInternalServerError {
-			controller.InternalError(w, r, c.h, errors.New(resp.Error))
-			return
-		}
-		c.h.RenderJSON(w, result.HTTPCode, resp)
+
+	switch res.HTTPCode {
+	case http.StatusInternalServerError:
```

Comment:
I'm not sure I follow why these cases behave differently?

Reply:
before I was rendering `struct IssueCodeResponse` which has:
`ErrorCode string 'json:"errorCode,omitempty"'`

We put error and errorCode on the responses which perhaps help document the possible responses. In practice we respond with:
`type ErrorReturn struct`
which has
`ErrorCode string 'json:"errorCode"'` with no `omitempty`

Reply:
the key difference here (other than the switch which I just thought looked better:
`c.h.RenderJSON(w, result.HTTPCode, resp)`
is now
`c.h.RenderJSON(w, result.HTTPCode, res.ErrorReturn)`